superbench/benchmarks/micro_benchmarks/cuda_decode_performance/AppDecPerf.cpp

```diff
@@ -0,0 +1,426 @@
+          std::vector<double> &vnLatency) {
+    std::vector<std::future<double>> decodeLatencyFutures;
+    ThreadPool threadPool(nThread);
+    // Enqueue the video decoding task into thread pool
+    auto start = std::chrono::high_resolution_clock::now();
+    for (int i = 0; i < files.size(); i++) {
+        auto filePath = files[i].c_str();
+        CheckInputFile(filePath);
+        decodeLatencyFutures.push_back(
+            threadPool.enqueue(DecodeVideo, vDec, filePath, &vnFrame[i], std::ref(vExceptionPtrs[i])));
+    }
+    // Wait until decoding tasks finished
+    for (int i = 0; i < files.size(); i++) {
+        auto decodeLatency = decodeLatencyFutures[i].get();
+        vnLatency.push_back(decodeLatency);
+        nTotalFrames += vnFrame[i];
+    }
+    // Calculated the metrics
+    return (std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now() - start)
+                .count()) /
+           1000.0f;
+}
+
+int main(int argc, char **argv) {
+    char szInFilePath[256] = "";
+    int iGpu = 0;
+    int nThread = 5;
+    int nTotalVideo = 100;
+    bool bSingle = false;
+    bool bHost = false;
+    std::string inputFilesListPath = "";
+    std::string outputFilePath = "";
+    std::vector<std::exception_ptr> vExceptionPtrs(nTotalVideo);
+    cudaVideoCodec codec = cudaVideoCodec_H264;
+    try {
+        // Parse the command line arguments
+        ParseCommandLine(argc, argv, szInFilePath, iGpu, nThread, nTotalVideo, bSingle, bHost, inputFilesListPath,
+                         outputFilePath, codec);
+        auto files = GenerateTotalFileList(inputFilesListPath, nTotalVideo, szInFilePath);
+
+        // Initialize and prepare the decoder context for each thread
+        std::vector<OptimizedNvDecoder *> vDec;
+        InitializeContext(vDec, iGpu, nThread, bSingle, bHost, codec);
+
+        // Decode all video with thread pool
+        std::vector<int> vnFrame(nTotalVideo);
+        int nTotalFrames = 0;
+        std::vector<double> vnLatency;
+
+        auto elapsedTime = run(vDec, nThread, files, vnFrame, vExceptionPtrs, &nTotalFrames, vnLatency);
+
+        // Calculate and output the raw data into file and metrics into stdout
+        double sum, mean, min, max, p50, p90, p95, p99;
+        std::tie(sum, mean, min, max, p50, p90, p95, p99) = CalLatencyMetrics(vnLatency);
+        if (outputFilePath.size() != 0) {
+            WriteRawData(vnLatency, vnFrame, outputFilePath);
+        }
+        std::cout << "Total Frames Decoded=" << nTotalFrames << " FPS=" << nTotalFrames / elapsedTime
+                  << " LatencyPerFrame=" << elapsedTime / nTotalFrames * 1000
+                  << " Mean Latency for each video=" << mean * 1000 << " P50 Latency=" << p50 * 1000
+                  << " P90 Latency=" << p90 * 1000 << " P95 Latency=" << p95 * 1000 << " P99 Latency=" << p99 * 1000
```

Comment:
I think we can output two-layer metrics: Hardware level (FPS and Frame Latency) and Data level (VideoFPS and VideoLatency)